pkgs/top-level/all-packages.nix

```diff
@@ -181,6 +181,26 @@ with pkgs;
       }
     '');
 
+  # libtool.m4 prior to version 2.4.7 invokes "/usr/bin/file", and is vendored into many ./configure scripts
+  replaceUsrBinFile =
```

Comment:
Maybe we could convert this to a setupHook in which case we would never need to call it manually? We would need to remove the exit part but it would make it a lot less work to maintain.

https://github.com/NixOS/nixpkgs/pull/166486/files#diff-7510b47653eab58ead900c36ffc4881682b881b87c0e7b7016d1f3947dee495cR50

Reply:
I think it would be logical to make this part of `fixLibtool` in `pkgs/stdenv/generic/setup.sh`.

Reply:
I would love to do that but I can't figure out how to thread the reference to ${buildPackages.file} down into `setup.sh`.  Also it would create a circular dependency in many cases, wouldn't it?

By the way, speaking of crazy circular dependencies, the `./configure` script that ships with `file-5.41.tar.gz` tries to call `/usr/bin/file`!  The package *ships with* a circular dependency!  Talk about crazy...

Reply:
> Maybe we could convert this to a setupHook

Thanks I will try that.

Reply:
> the `./configure` script that ships with `file-5.41.tar.gz` tries to call `/usr/bin/file`! The package _ships with_ a circular dependency!

> I would love to do that but I can't figure out how to thread the reference to `${buildPackages.file}` down into `setup.sh`. Also even if I could do that, it would create a circular dependency in many cases, wouldn't it?

I should add that these are two arguments for considering putting `file` in the bootstrap-tools.  I know that adding more stuff to the bootstrap-tools is never an easy decision, but when you have packages that depend on themselves (like a C compiler) and which 99% of the ecosystem has a vendored dependency upon, well... that's sort of what the bootstrap-files is for.

Then we could have `fixLibtool()` replace `/usr/bin/file` with `file`, since the bootstrap-tools `bin` directory is always in the `$PATH`.  So there wouldn't be any need to thread the package reference down into there through the nixlang code.

It's ugly, but so is the problem that it fixes.

Reply:
See #168413 for a much, much simpler solution that simply (a) patches configure scripts unconditionally and (b) includes `file` in stdenv's `common-path.nix`.

I'm running a rebuild-the-world on that PR, it'll take all day.